pkgs/development/libraries/mpfi/default.nix

```diff
@@ -1,23 +1,28 @@
-{lib, stdenv, fetchurl, autoreconfHook, texinfo, mpfr}:
+{lib, stdenv, fetchFromGitLab, autoreconfHook, texinfo, mpfr}:
 stdenv.mkDerivation rec {
   pname = "mpfi";
   version = "1.5.4";
-  file_nr = "38111";
 
-  src = fetchurl {
-    # NOTE: the file_nr is whats important here. The actual package name (including the version)
-    # is ignored. To find out the correct file_nr, go to https://gforge.inria.fr/projects/mpfi/
-    # and click on Download in the section "Latest File Releases".
-    url = "https://gforge.inria.fr/frs/download.php/file/${file_nr}/mpfi-${version}.tgz";
-    sha256 = "sha256-Ozk4WV1yCvF5c96vcnz8DdQcixbCCtwQOpcPSkOuOlY=";
+  src = fetchFromGitLab {
+    domain = "gitlab.inria.fr";
+    owner = "mpfi";
+    repo = "mpfi";
+
+    # Apparently there is an upstream off-by-one-commit error in tagging
+    #rev = version;
+    rev = "feab26bc54529417af983950ddbffb3a4c334d4f";
```

Comment:
Maybe conditionalize this, so future updates won't accidently use the same commit?

Reply:
Actually I missed a very good point of your proposal: a conditional has other benefits (e.g. auto-updates would happen even if suboptimally, and can get fixed on top of the update PR). I do think that a more localised mess could be devised though, let me try.

Reply:
(Done)